rezervo/integrations/fsc/booking.py

```diff
@@ -0,0 +1,229 @@
+    user_details = fetch_user_details(auth_session)
+    if isinstance(auth_session, AuthenticationError):
+        return auth_session
+    return f"https://fsc.no/api/v1/auth/customers/{user_details['id']}/bookings/groupactivities"
+
+
+def find_fsc_class_by_id(
+    integration_user: IntegrationUser, config: ConfigValue, class_id: str
+) -> Union[RezervoClass, None, BookingError, AuthenticationError]:
+    print(f"Searching for class by id: {class_id}")
+    fsc_schedule = fetch_fsc_schedule()
+    if fsc_schedule is None:
+        err.log("Class get request failed")
+        return BookingError.ERROR
+    fsc_class = next(
+        (c for c in fsc_schedule if c.id == int(class_id)),
+        None,
+    )
+    if fsc_class is None:
+        return BookingError.CLASS_MISSING
+    return rezervo_class_from_fsc_class(fsc_class)
+
+
+def find_fsc_class(
+    _class_config: Class,
+) -> Union[RezervoClass, BookingError, AuthenticationError]:
+    print(f"Searching for class matching config: {_class_config}")
+    schedule = fetch_fsc_schedule()
+    if schedule is None:
+        err.log("Schedule get request denied")
+        return BookingError.ERROR
+    if not 0 <= _class_config.weekday < len(WEEKDAYS):
+        err.log(f"Invalid weekday number ({_class_config.weekday=})")
+        return BookingError.MALFORMED_SEARCH
+
+    result = None
+    for c in schedule:
+        if c.groupActivityProduct.id != _class_config.activity:
+            continue
+        start_time = datetime.fromisoformat(c.duration.start.replace("Z", "")[:-4])
+        utc_start_time = pytz.timezone("UTC").localize(start_time)
+        localized_start_time = utc_start_time.astimezone(pytz.timezone("Europe/Oslo"))
+        time_matches = (
+            localized_start_time.hour == _class_config.time.hour
+            and localized_start_time.minute == _class_config.time.minute
+        )
+        if not time_matches:
+            print(f"Found class, but start time did not match: {c}")
+            result = BookingError.INCORRECT_START_TIME
+            continue
+        if localized_start_time.weekday() != _class_config.weekday:
+            print(f"Found class, but weekday did not match: {c}")
+            result = BookingError.MISSING_SCHEDULE_DAY
+            continue
+        search_feedback = f'Found class: "{c.name}"'
+        if len(c.instructors) > 0:
+            search_feedback += (
+                f" with {format_name_list_to_natural([i.name for i in c.instructors])}"
+            )
+        else:
+            search_feedback += " (missing instructor)"
+        search_feedback += f" at {c.duration.start}"
+        print(search_feedback)
+        return rezervo_class_from_fsc_class(c)
+    err.log("Could not find class matching criteria")
+    if result is None:
+        result = BookingError.CLASS_MISSING
+    return result
+
+
+def book_fsc_class(auth_session: Session, class_id: int) -> bool:
```

Comment:
The naming of `book_fsc_class` and `book_class` could be better (this is a problem for the Sit integration as well)